tools/apps/src/lib.rs

```diff
@@ -0,0 +1,900 @@
+    pub max_data: u64,
+    pub max_stream_data: u64,
+    pub max_streams_bidi: u64,
+    pub max_streams_uni: u64,
+    pub dump_packet_path: Option<String>,
+    pub no_grease: bool,
+}
+
+/// Creates a new `CommonArgs` structure using the provided [`Docopt`].
+///
+/// The `Docopt` usage String needs to include the following:
+///
+/// --http-version VERSION   HTTP version to use
+/// --max-data BYTES         Connection-wide flow control limit.
+/// --max-stream-data BYTES  Per-stream flow control limit.
+/// --max-streams-bidi STREAMS  Number of allowed concurrent streams.
+/// --max-streams-uni STREAMS   Number of allowed concurrent streams.
+/// --dump-packets PATH         Dump the incoming packets as files in the
+/// given directory. --no-grease                 Don't send GREASE.
+///
+/// [`Docopt`]: https://docs.rs/docopt/1.1.0/docopt/
+impl Args for CommonArgs {
+    fn with_docopt(docopt: &docopt::Docopt) -> Self {
+        let args = docopt.parse().unwrap_or_else(|e| e.exit());
+
+        let http_version = args.get_str("--http-version");
+        let alpns = match http_version {
+            "HTTP/0.9" => alpns::length_prefixed(&alpns::HTTP_09),
+
+            "HTTP/3" => alpns::length_prefixed(&alpns::HTTP_3),
+
+            "all" => [
+                alpns::length_prefixed(&alpns::HTTP_3),
+                alpns::length_prefixed(&alpns::HTTP_09),
+            ]
+            .concat(),
+
+            _ => panic!("Unsupported HTTP version"),
+        };
+
+        let max_data = args.get_str("--max-data");
+        let max_data = u64::from_str_radix(max_data, 10).unwrap();
+
+        let max_stream_data = args.get_str("--max-stream-data");
+        let max_stream_data = u64::from_str_radix(max_stream_data, 10).unwrap();
+
+        let max_streams_bidi = args.get_str("--max-streams-bidi");
+        let max_streams_bidi = u64::from_str_radix(max_streams_bidi, 10).unwrap();
+
+        let max_streams_uni = args.get_str("--max-streams-uni");
+        let max_streams_uni = u64::from_str_radix(max_streams_uni, 10).unwrap();
+
+        let dump_packet_path = if args.get_str("--dump-packets") != "" {
+            Some(args.get_str("--dump-packets").to_string())
+        } else {
+            None
+        };
+
+        let no_grease = args.get_bool("--no-grease");
+
+        CommonArgs {
+            alpns,
+            max_data,
+            max_stream_data,
+            max_streams_bidi,
+            max_streams_uni,
+            dump_packet_path,
+            no_grease,
+        }
+    }
+}
+
+pub struct PartialResponse {
+    pub body: Vec<u8>,
+
+    pub written: usize,
+}
+
+pub struct Client {
+    pub conn: std::pin::Pin<Box<quiche::Connection>>,
+
+    pub http_conn: Option<Box<dyn crate::HttpConn>>,
+
+    pub partial_responses: std::collections::HashMap<u64, PartialResponse>,
+}
+
+pub type ClientMap = HashMap<Vec<u8>, (net::SocketAddr, Client)>;
+
+// Makes a buffered writer for a resource with a target URL. The
```

Comment:
```suggestion
/// Makes a buffered writer for a resource with a target URL. The
```
Also, the rest of the lines. Also, should have first line until first period, then empty line, then rest of the comment.